### PR TITLE
Add unit tests from jasp-desktop and fix missing dependencies

### PR DESCRIPTION
**Pull request by TimKDJ**

### CODE REVIEW

**Comment by TimKDJ** [comment]
I'm going to make a bunch of these PR's.
If you could gloss over it for obvious errors that'd be great @vandenman

**Comment by AlexanderLyNL** [comment]
Do the following files need to be part of it?

A .Rbuildignore (2)
A .gitignore (54)

**Comment by TimKDJ** [comment]
Yes, because otherwise if people make any changes they'll get a bunch of unnecessary files in their PR

File: jaspTTests.Rproj [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+AlwaysSaveHistory: Default
+
+EnableCodeIndexing: Yes
+UseSpacesForTab: Yes
+NumSpacesForTab: 2
+Encoding: UTF-8
+
+RnwWeave: Sweave
+LaTeX: pdfLaTeX
+
```
Comment (vandenman) [inline jaspTTests.Rproj:14]:
```suggestion

AutoAppendNewline: Yes
StripTrailingWhitespace: Yes

```
that would ensure there are no unnessesary files changed due to trailing whitespace and Windows vs unix line endings.

File: tests/testthat/test.csv [hunk 2]
```diff
@@ -0,0 +1,101 @@
+,contNormal,contGamma,contBinom,contExpon,contWide,contNarrow,contOutlier,contcor1,contcor2,facGender,facExperim,facFive,facFifty,facOutlier,debString,debMiss1,debMiss30,debMiss80,debMiss99,debBinMiss20,debNaN,debNaN10,debInf,debCollin1,debCollin2,debCollin3,debEqual1,debEqual2,debSame
```
Comment (vandenman) [inline tests/testthat/test.csv:1]: So... every module is going to have a copy of the debug dataset?

I would do that a little bit differently.

perhaps:

1. a specific package for this (called`jaspDatasets`?). A very rigorous solution, but probably overkill.
2. make the debug data set live inside `jasptools`.

**Review by vandenman — CHANGES_REQUESTED** [review]
I didn't look at test figures and test R files. Mostly looks good to me aside from the two small comments.

File: tests/figs/jasp-deps.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+- jaspGraphs: 0.5
```
Comment (vandenman) [inline tests/figs/jasp-deps.txt:1]: does this mean that if I bump the jaspGraphs version I have to increment this file in every module?

File: tests/testthat/test.csv (hunk 2 above)
Comment (TimKDJ) [reply tests/testthat/test.csv:1]: jaspTools fetches all the regular datasets from jasp-desktop Resources, except for unit test specific ones.
I could include `test.csv` as well I guess, but I wasn't sure if people would want to edit this file.

File: tests/figs/jasp-deps.txt (hunk 3 above)
Comment (TimKDJ) [reply tests/figs/jasp-deps.txt:1]: Yes. We can't guarantee the jaspGraphs version developers use in new modules, so we can't do this centrally.
Besides all plots need to be verified after a jaspGraphs bump anyhow.

**Pull request merged**